<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>StructFieldDialog</class>
 <widget class="QDialog" name="StructFieldDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>450</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Struct Field Configuration</string>
  </property>
  <property name="styleSheet">
   <string notr="true">/* Dark theme stylesheet */
QDialog, QWidget {
    background-color: #1e1e1e;
    color: #f0f0f0;
}

QLabel {
    color: #f0f0f0;
}

QFrame {
    background-color: #2d2d2d;
    border-radius: 5px;
}

QPushButton {
    background-color: #0d6efd;
    color: white;
    border: none;
    padding: 6px 12px;
    border-radius: 4px;
}

QPushButton:hover {
    background-color: #0b5ed7;
}

QPushButton:pressed {
    background-color: #0a58ca;
}

QLineEdit, QPlainTextEdit, QSpinBox, QDoubleSpinBox, QComboBox {
    background-color: #363636;
    border: 1px solid #505050;
    color: #f0f0f0;
    padding: 4px;
    border-radius: 3px;
}

QLineEdit:focus, QPlainTextEdit:focus, QSpinBox:focus, QDoubleSpinBox:focus, QComboBox:focus {
    border: 1px solid #0d6efd;
}

QGroupBox {
    border: 1px solid #505050;
    border-radius: 5px;
    margin-top: 1ex;
    font-weight: bold;
}

QGroupBox::title {
    subcontrol-origin: margin;
    subcontrol-position: top center;
    padding: 0 5px;
    color: #f0f0f0;
}

QDialogButtonBox {
    background-color: #1e1e1e;
    border-top: 1px solid #505050;
    padding-top: 15px;
}</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QFormLayout" name="form_layout">
     <item row="0" column="0">
      <widget class="QLabel" name="name_label">
       <property name="text">
        <string>Field Name:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="name_edit"/>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="type_group">
     <property name="title">
      <string>Field Type</string>
     </property>
     <layout class="QVBoxLayout" name="type_layout">
      <item>
       <layout class="QHBoxLayout" name="radio_layout">
        <item>
         <widget class="QRadioButton" name="basic_radio">
          <property name="text">
           <string>Basic Type</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="array_radio">
          <property name="text">
           <string>Array Type</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="type_selection_layout">
        <item>
         <widget class="QComboBox" name="type_combo">
          <item>
           <property name="text">
            <string>bool_t</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>uint8</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>uint16</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>uint32</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>uint64</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>sint8</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>sint16</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>sint32</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>sint64</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>char_t</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>float32</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>float64</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="array_layout">
        <item>
         <widget class="QLabel" name="array_type_label">
          <property name="visible">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Array of:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="array_display">
          <property name="visible">
           <bool>false</bool>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="configure_array_button">
          <property name="visible">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Configure...</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QFormLayout" name="description_layout">
     <item row="0" column="0">
      <widget class="QLabel" name="description_label">
       <property name="text">
        <string>Description:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="description_edit"/>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>